sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -201,6 +201,66 @@ mod checked {
         (inner, effects, execution_result)
     }
 
+    #[instrument(
+        name = "commit_transaction_without_execution",
+        level = "debug",
+        skip_all
+    )]
+    pub fn commit_transaction_without_execution<Mode: ExecutionMode>(
```

Comment:
Instead of a new function that has quite some identical code, could we not pass a failure reason to execute_transaction_to_effects and skip execution if specified? That way we can even merge the failure due to deleted shared object and due to cert denied.

Reply:
Yeah, I intend to have the cert denied take this new code path. Having a separate code path is my preference in code style